library/summary.ml

```diff
@@ -124,6 +128,28 @@ let init_summaries () =
 
 let nop () = ()
 
+(** Summary projection *)
+let project_typed_summary { summaries } tag =
+  let id = Dyn.repr tag in
+  let hash = String.hash id in
+  let state = List.assoc hash summaries in
```

Comment:
Use `CList.assoc_f`. Functions relying on generic equality should be avoided.

Reply:
How about turning `summaries` into map? I don't see why it makes sense for it to be a list. In fact, a duplicate entry should be an error, it looks to me.

Moreover, projecting from it may become a common operation in the future (not a worry I think as we are talking about a per-command cost)